src/formatDate.js

```diff
@@ -8,7 +8,31 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const separatorFrom = fromFormat[3];
+  const separatorTo = toFormat[3];
+  const dateParts = date.split(separatorFrom);
+  const dateMap = {};
+
+  fromFormat.slice(0, 3).forEach((part, index) => {
+    dateMap[part] = dateParts[index];
+  });
+
+  const convertToYYYY = (yy) => (parseInt(yy, 10) < 30 ? `20${yy}` : `19${yy}`);
```

Comment:
Consider checking for the length of `yy` before parsing it. Although the input is assumed to be correct, ensuring that `yy` is exactly two digits long could prevent potential bugs if the function is used in a different context in the future.